src/inmanta/server/services/orchestrationservice.py

```diff
@@ -517,6 +624,38 @@ async def _put_version(
         *,
         connection: asyncpg.connection.Connection,
     ) -> None:
+        """
+        :param rid_to_resource: This parameter should contain all the resources when a full compile is done.
+                                When a partial compile is done, it should contain all the resources that belong to the
+                                updated resource sets or the shared resource sets.
+        :param unknowns: This parameter should contain all the unknowns for all the resources in the new version of the model.
+                         Also the unknowns for resources that are not present in rid_to_resource.
+        :param partial_base_version: When a partial compile is done, this parameter contains the version of the
+                                     configurationmodel this partial compile was based on. Otherwise this parameter should be
+                                     None.
+        :param removed_resource_sets: When a partial compile is done, this parameter should indicate the names of the resource
+                                      sets that are removed by the partial compile. When no resource sets are removed by
+                                      a partial compile or when a full compile is done, this parameter can be set to None.
+
+        Pre-conditions:
+            * The requires and provides relationships of the resources in rid_to_resource must to be set correctly.
```

Comment:
```suggestion
            * The requires and provides relationships of the resources in rid_to_resource must be set correctly. For a partial export, this means it is assumed to be valid with respect to all absolute constraints that apply to partial compiles. Constraints that are relative to the base version will be verified by this method.
```
It wasn't clear to me what this meant exactly. I think this is the essence of it?